service/matching/handler.go

```diff
@@ -37,38 +37,53 @@ import (
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/persistence"
 	"github.com/uber/cadence/common/quotas"
-	"github.com/uber/cadence/common/service"
+	"github.com/uber/cadence/common/resource"
 )
 
 var _ matchingserviceserver.Interface = (*Handler)(nil)
 
-// Handler - Thrift handler inteface for history service
+// Handler - Thrift handler interface for history service
 type Handler struct {
-	taskPersistence persistence.TaskManager
-	metadataMgr     persistence.MetadataManager
-	engine          Engine
-	config          *Config
-	metricsClient   metrics.Client
-	startWG         sync.WaitGroup
-	domainCache     cache.DomainCache
-	rateLimiter     quotas.Limiter
-	service.Service
+	resource.Resource
+
+	taskMgr       persistence.TaskManager
```

Comment:
You don't need to keep taskMgr, metadataMgr, domainCache, config here.

Reply:
done